extensions/amp-analytics/0.1/variables.js

```diff
@@ -317,6 +318,17 @@ export function variableServiceForDoc(elementOrAmpDoc) {
   return getServiceForDoc(elementOrAmpDoc, 'amp-analytics-variables');
 }
 
+/**
+ * @param {!Element|!ShadowRoot|!../../../src/service/ampdoc-impl.AmpDoc} elementOrAmpDoc
+ * @return {!Promise<!VariableService>}
+ */
+export function variableServicePromiseForDoc(elementOrAmpDoc) {
```

Comment:
do we still need the other sync method?

Reply:
The other classes created by `amp-analytics` (linker, cookie-writer, requests etc) are still using the sync method. This is ok because they get created after amp-analytics resolves the promise.  I thought about changing them all but I think it would make the code harder to follow in these other files.